File: adminpanel/src/services/AnalyticsService.js
// ---------------- ANALYTICS APIs (FIXED) ----------------
import api from "./CustomAxiosInstance";



export const getMonthlySales = async () => {
  const resp = await api.get("/api/analytics/sales/monthly");
  return resp.data;
};

export const getMonthlyStockHistory = async () => {
  const resp = await api.get("/api/analytics/stock/history/monthly");
  return resp.data;
};

export const getTopSellingFoods = async () => {
  const resp = await api.get("/api/analytics/top-selling");
  return resp.data;
};

export const getLowStockFoods = async () => {
  const resp = await api.get("/api/analytics/low-stock");
  return resp.data;
};

File: foodapi/src/main/java/com/nisanth/foodapi/service/AnalyticsService.java
package com.nisanth.foodapi.service;

import com.nisanth.foodapi.entity.StockLogEntity;
import com.nisanth.foodapi.io.LowStockDTO;
import com.nisanth.foodapi.io.MonthlySalesDTO;
import com.nisanth.foodapi.io.StockHistoryDTO;
import com.nisanth.foodapi.io.TopSellingDTO;
import com.nisanth.foodapi.repository.AnalyticsRepository;
import com.nisanth.foodapi.repository.OrderRepository;
import com.nisanth.foodapi.repository.StockLogRepository;
import com.nisanth.foodapi.repository.FoodRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class AnalyticsService {

    private final StockLogRepository stockLogRepository;
    private final AnalyticsRepository analyticsRepository;
    private final FoodRepository foodRepository;
    private final OrderRepository orderRepository;

    public List<StockLogEntity> getAllStockLogs() {
        return stockLogRepository.findAllByOrderByTimestampDesc();
    }

    // ------------------------------
    // MONTHLY SALES
    // ------------------------------
    public List<MonthlySalesDTO> getMonthlySales() {
        return analyticsRepository.getMonthlySales();
    }

    // ------------------------------
    // MONTHLY STOCK HISTORY
    // ------------------------------
    public List<StockHistoryDTO> getMonthlyStockHistory() {
        return stockLogRepository.getMonthlyStockHistory();
    }

    // ------------------------------
    // TOP SELLING ITEMS
    // ------------------------------
    public List<TopSellingDTO> getTopSellingFoods() {
        return orderRepository.getTopSellingFoods();
    }

    // ------------------------------
    // LOW STOCK ITEMS
    // ------------------------------
    public List<LowStockDTO> getLowStockItems() {
        return foodRepository.getLowStockItems();
    }
}
